pkg/agent/proxy/proxier.go

```diff
@@ -1165,7 +1165,7 @@ func endpointSliceAPIAvailable(k8sClient clientset.Interface) (bool, error) {
 		return false, fmt.Errorf("error getting server resources for GroupVersion %s: %v", discovery.SchemeGroupVersion.String(), err)
 	}
 	for _, resource := range resources.APIResources {
-		if resource.Name == "endpointslice" {
```

Comment:
I don't remember why I used singular name for this and if I tested it with a cluster with available EndpointSlice API. Anyway change to match Kind which is more visible in API response.